packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -72,17 +72,31 @@ export const SelectionHandler = (
     if (evt.timeStamp - (lastPointerDown?.timeStamp || evt.timeStamp) < 1000 && !currentTarget)
       onSelectStart(lastPointerDown);
 
-    if (sel.isCollapsed || !isLeftClick || !currentTarget) return;
+    // The selection isn't active -> bail out from selection change processing
+    if (!currentTarget)
+      return;
+
+    /**
+     * The selection range got collapsed during the selecting process.
+     * The previously created annotation isn't relevant anymore and can be discarded
+     *
+     * @see https://github.com/recogito/text-annotator-js/issues/139
+     */
+    if (sel.isCollapsed && store.getAnnotation(currentTarget.annotation)) {
+      selection.clear();
+      store.deleteAnnotation(currentTarget.annotation);
+      return;
+    }
```

Comment:
I intentionally don't reset the `currentTarget` here, because during the mouse selection cycle, the `selectstart` won't be dispatched another time to create a new target instance.